packages/office-ui-fabric-react/src/components/Dropdown/Dropdown.base.tsx

```diff
@@ -1036,33 +1035,36 @@ export class DropdownBase extends React.Component<IDropdownInternalProps, IDropd
     const { isOpen } = this.state;
     const disabled = this._isDisabled();
 
-    if (!disabled && !(this._isOpenedByKeyboardFocus && isOpen)) {
+    if (!disabled && !this._shouldOpenOnFocus()) {
```

Comment:
I would add a comment here about why `_shouldOpenOnFocus` is called. It might be confusing as to why that's called in `_onDropdownClick`